pkg/api/owners.go

```diff
@@ -0,0 +1,687 @@
+}
+
+func (f *FieldOwners) ClaimMemKernelLimit(plugin string) error {
+	return f.ClaimSimple(Field_MemKernelLimit.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimMemTCPLimit(plugin string) error {
+	return f.ClaimSimple(Field_MemTCPLimit.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimMemSwappiness(plugin string) error {
+	return f.ClaimSimple(Field_MemSwappiness.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimMemDisableOomKiller(plugin string) error {
+	return f.ClaimSimple(Field_MemDisableOomKiller.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimMemUseHierarchy(plugin string) error {
+	return f.ClaimSimple(Field_MemUseHierarchy.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPUShares(plugin string) error {
+	return f.ClaimSimple(Field_CPUShares.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPUQuota(plugin string) error {
+	return f.ClaimSimple(Field_CPUQuota.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPUPeriod(plugin string) error {
+	return f.ClaimSimple(Field_CPUPeriod.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPURealtimeRuntime(plugin string) error {
+	return f.ClaimSimple(Field_CPURealtimeRuntime.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPURealtimePeriod(plugin string) error {
+	return f.ClaimSimple(Field_CPURealtimePeriod.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPUSetCPUs(plugin string) error {
+	return f.ClaimSimple(Field_CPUSetCPUs.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCPUSetMems(plugin string) error {
+	return f.ClaimSimple(Field_CPUSetMems.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimPidsLimit(plugin string) error {
+	return f.ClaimSimple(Field_PidsLimit.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimHugepageLimit(size, plugin string) error {
+	return f.ClaimCompound(Field_HugepageLimits.Key(), size, plugin)
+}
+
+func (f *FieldOwners) ClaimBlockioClass(plugin string) error {
+	return f.ClaimSimple(Field_BlockioClass.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimRdtClass(plugin string) error {
+	return f.ClaimSimple(Field_RdtClass.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimCgroupsUnified(key, plugin string) error {
+	return f.ClaimCompound(Field_CgroupsUnified.Key(), key, plugin)
+}
+
+func (f *FieldOwners) ClaimCgroupsPath(plugin string) error {
+	return f.ClaimSimple(Field_CgroupsPath.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimOomScoreAdj(plugin string) error {
+	return f.ClaimSimple(Field_OomScoreAdj.Key(), plugin)
+}
+
+func (f *FieldOwners) ClaimRlimit(typ, plugin string) error {
+	return f.ClaimCompound(Field_Rlimits.Key(), typ, plugin)
+}
+
+func (f *FieldOwners) ClearCompound(field int32, key, plugin string) {
+	m, ok := f.Compound[field]
+	if !ok {
+		m = NewCompoundFieldOwners()
+		f.Compound[field] = m
+	}
+
+	m.Owners[key] = MarkForRemoval(plugin)
+}
+
+func (f *FieldOwners) ClearSimple(field int32, plugin string) {
+	f.Simple[field] = MarkForRemoval(plugin)
+}
+
+func (f *FieldOwners) ClearAnnotation(key, plugin string) {
```

Comment:
Bringing over https://github.com/containerd/nri/pull/163#discussion_r2070959731

> These `Clear` functions probably deserve a bit of a comment explaining the situations in which they should be called. It's not immediately obvious why we'd clear ownership for an existing field.

It looks like the generic answer is "when an adjustment deletes the field".

Reply:
None of these need to be called directly by a plugins. These are used internally in pkg/adaptation for ownership tracking and conflict detection. Added clarifying comments to the beginning of owners.go to that effect.